src/aes.rs

```diff
@@ -0,0 +1,276 @@
+use super::Error;
+use crate::{
+    create_signed_header,
+    icl_header_v4::{self, v4document_header::edek_wrapper::Edek},
+    EncryptedDocument, EncryptedPayload, EncryptionKey, PlaintextDocument, MAGIC, V0,
+};
+use aes_gcm::{aead::Aead, aead::Payload, AeadCore, Aes256Gcm, KeyInit, Nonce};
+use bytes::Bytes;
+use rand::{CryptoRng, RngCore};
+
+type Result<T> = core::result::Result<T, super::Error>;
+const DETATCHED_HEADER_LEN: usize = 5;
```

Comment:
```suggestion
const DETACHED_HEADER_LEN: usize = 5;
```

Reply:
I just did a search for "detatch" in this file... I'm not gonna comment on each one :)